qiskit_experiments/framework/base_experiment.py

```diff
@@ -265,31 +265,92 @@ def _finalize(self):
         """
         pass
 
-    def _run_jobs(self, circuits: List[QuantumCircuit], **run_options) -> List[Job]:
-        """Run circuits on backend as 1 or more jobs."""
+    def _max_circuits(self, backend: Backend = None):
+        """
+        Calculate the maximum number of circuits per job for the experiment.
+        """
+
+        if backend is None:
+            if self.backend is None:
+                raise QiskitError("A backend must be provide.")
+            backend = self.backend
+        else:
+            self._set_backend(backend)
```

Comment:
I don't think this should be setting the backend. If the user is just passing in different backends to `job_info()` expecting to get information on the number of jobs, they might be surprised to find this was actually also setting the actual backend the experiment will be run on.

Reply:
yeah, I made a Mistake there I guess `self.backend = backend` will work fine.